packages/xo-server/src/api/host.mjs

```diff
@@ -136,13 +137,33 @@ export async function restart({
     const pool = this.getObject(host.$poolId, 'pool')
     const master = this.getObject(pool.master, 'host')
     const hostRebootRequired = host.rebootRequired
-    if (hostRebootRequired && host.id !== master.id && host.version === master.version) {
-      throw incorrectState({
-        actual: hostRebootRequired,
-        expected: false,
-        object: master.id,
-        property: 'rebootRequired',
-      })
+
+    if (hostRebootRequired && host.id !== master.id) {
+      const throwError = () =>
+        incorrectState({
+          actual: hostRebootRequired,
+          expected: false,
+          object: master.id,
+          property: 'rebootRequired',
+        })
+
+      if (semver.lt(master.version, host.version)) {
+        log.error(`master version (${master.version}) is older than the host version (${host.version})`, {
+          masterId: master.id,
+          hostId: host.id,
+        })
+        throwError()
+      }
+
+      const isMasterVersionEqual = semver.eq(master.version, host.version)
+      if (isMasterVersionEqual && (await this.getXapi(host).listMissingPatches(master._xapiId)).length > 0) {
+        log.error('master has missing patches', { masterId: master.id })
+        throwError()
```

Comment:
here, I think we can have an incorrectState error with rebootRequired in the right value , maybe we could have a incorrectState on host missingpatches length ?

Reply:
I'm not sure to understand. You suggest throwing an `incorrectState` error and passing this as argument?
```
{
  property: 'missingPatches',
  expected: 0,
  actual: missingPatches.length,
  object: master.id
}
```

Reply:
ok, I understand, since hostRebootRequired is trueish here (line 141 ) , we can't have incorrectState throw an error with actual and expected equal